src/bscPlugin/validation/ScopeValidator.ts

```diff
@@ -27,181 +27,133 @@ const platformComponentNames = new Set(Object.values(components).map(x => x.name
  */
 export class ScopeValidator {
 
-    private events: OnScopeValidateEvent[] = [];
+    /**
+     * The event currently being processed. This will change multiple times throughout the lifetime of this validator
+     */
+    private event: OnScopeValidateEvent;
 
     public processEvent(event: OnScopeValidateEvent) {
-        this.events.push(event);
-        event.scope.linkSymbolTable();
```

Comment:
Symbol table linking/unlinking happens at the program event level, we don't need to do it here anymore.